setup.py

```diff
@@ -24,5 +24,5 @@
         'Intended Audience :: Developers',
         'Intended Audience :: Science/Research'
     ],
-    install_requires=['nbconvert>=5.4.0', 'nbformat', 'jinja2']
+    install_requires=['nbconvert>=5.4.0', 'nbformat', 'jinja2', 'junit_xml']
```

Comment:
I have some misgivings about adding `junit_xml` as a dependency. Perhaps it can be "optional" in that if the writing-out is disabled (say, if you take my suggestion above, by having the output file be `None`), then `junit_xml` is not necessary (at least if there are appropriate try/excepts around the import - I can point to some examples of this pattern if you're not familiar with it).  Does that seem possible?

Reply:
@eteq Problem is that use of this module is segregated throughout the program. It is needed in execute to create a test case and it is needed at the end of process notebooks to create a test suite after all notebooks are finished and all test cases are finished. Python imports (unless done globally) are local to the function (as I just found out implementing this). In order to make this import optional with a argument, we would have to do conditional imports in all functions that use it (messy in my opinion) unless you know of another way? I'm not the most experienced with python @eteq